legal-api/src/legal_api/models/document.py

```diff
@@ -43,6 +44,8 @@ class Document(db.Model):
     id = Column(db.Integer, primary_key=True)
     type = Column('type', String(30), nullable=False)
     file_key = Column('file_key', String(100), nullable=False)
+    file_name = Column('file_name', String(200), nullable=False)
```

Comment:
New fields (well not new I think this was recently removed)

Reply:
Why do you need this? We are not saving the file name in any file upload

Reply:
Requirement here:

![image](https://user-images.githubusercontent.com/3484109/235805548-5fcfbee7-a900-47f6-ae15-f8e78d430035.png)

and here:
![image](https://user-images.githubusercontent.com/3484109/235805591-8f8208f1-bb64-473c-a698-b3aa11bf4783.png)

Reply:
Is it a user uploaded file name or system derived file name?

Reply:
We don't save file name in IA as well. User cannot download the file they uploaded instead they can access Certified copy (Certified Memorandum, Certified Rules).

Reply:
![image](https://user-images.githubusercontent.com/64869009/235809070-78aac09f-fef1-4426-92f0-389b65ca1384.png)

Reply:
Looks like it's saved to me - at least for CO-OPS. 

It needs to be saved in there for DRAFTS I believe? 

Yup that list is related to a specific filing and not a business. I need an endpoint that gives me the current state of the business.

Reply:
Right. For draft we do save the file name in json. For the current state of business you can generate the file name based on document `type`.
https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/models/document.py#L28

Reply:
The file name is relevant when it's in draft. Once processed its going to be (Certified Memorandum, Certified Rules).

Reply:
something like this:
```
if documentType == COOP_RULES:
    return "Rules of Association - {business_name}.pdf"
 ```
Ok let me just check with Ethan

Reply:
```
                self.create_report_object(
                    'Certified Rules',
                    self.get_general_filename('Certified Rules'),
                    DocumentMetaService.ReportType.CERTIFIED_RULES.value
                ),
                self.create_report_object(
                    'Certified Memorandum',
                    self.get_general_filename('Certified Memorandum'),
                    DocumentMetaService.ReportType.CERTIFIED_MEMORANDUM.value
                )
```

Ahh I see already a pattern established

Reply:
I've tried to reuse the same code that generates those names -^